File: lib/src/model/output_audio_device_info.dart
/// Kind of the [OutputAudioDeviceInfo].
enum OutputAudioDeviceInfoKind {
  /// Audio will be played on the ear speaker.
  earSpeaker,

  /// Audio will be played on the speakerphone.
  speakerphone,

  /// Audio will be played on some audio device which can't be assigned to the
  /// other [OutputAudioDeviceInfoKind]. Speakers in the browser, for example.
  unknown,
}

/// Information about some output audio device.
class OutputAudioDeviceInfo {
  /// Create a [OutputAudioDeviceInfo] based on the provided properties.
  OutputAudioDeviceInfo(this.deviceId, this.label, this.kind);

  /// Creates a [OutputAudioDeviceInfo] basing on the [Map] received from the
  /// native side.
  OutputAudioDeviceInfo.fromMap(dynamic map) {
    deviceId = map['deviceId'];
    label = map['label'];
    kind = OutputAudioDeviceInfoKind.values[map['kind']];
  }
  
  /// Identifier of the represented device.
  late String deviceId;
  
  /// Human-readable device description (for example, "Ear speaker").
  late String label;
  
  /// Kind of the device.
  late OutputAudioDeviceInfoKind kind;
}

File: android/src/main/kotlin/com/cloudwebrtc/webrtc/model/OutputAudioDeviceInfo.kt
package com.cloudwebrtc.webrtc.model

/**
 * Kind of the [OutputAudioDeviceInfo].
 */
enum class OutputAudioDeviceKind(val value: Int) {
    /**
     * Audio will be played on the ear speaker.
     */
    EAR_SPEAKER(0),

    /**
     * Audio will be played on the speakerphone.
     */
    SPEAKERPHONE(1),

    /**
     * Audio will be played on the bluetooth headset if it possible.
     */
    BLUETOOTH_HEADSET(2);
}

/**
 * Information about some output audio device.
 *
 * @property deviceId  Identifier of the represented device.
 * @property label  Human-readable device description (for example, "Ear speaker").
 * @property kind  Kind of the device.
 */
data class OutputAudioDeviceInfo(
    val deviceId: String,
    val label: String,
    val kind: OutputAudioDeviceKind
) {
    /**
     * Converts this [MediaDeviceInfo] into a [Map] which can be returned to the
     * Flutter side.
     */
    fun asFlutterResult(): Map<String, Any> = mapOf(
        "deviceId" to deviceId,
        "label" to label,
        "kind" to kind.value
    )
}
